### PR TITLE
Adopt webworker search for local files in browser

### PR DESCRIPTION
**Pull request by JacksonKearl**
Resolves https://github.com/microsoft/vscode-dev/issues/264

### CODE REVIEW

File: src/vs/platform/files/browser/htmlFileSystemProvider.ts [hunk 1]
```diff
@@ -365,7 +365,7 @@ export class HTMLFileSystemProvider implements IFileSystemProviderWithFileReadWr
 		}
 	}
 
-	private async getDirectoryHandle(resource: URI): Promise<FileSystemDirectoryHandle | undefined> {
+	async getDirectoryHandle(resource: URI): Promise<FileSystemDirectoryHandle | undefined> {
```
Comment (JacksonKearl) [inline src/vs/platform/files/browser/htmlFileSystemProvider.ts:368]: cc @bpasero we need the raw `FileSystemDirectoryHandle` on the webworker side to ensure minimum possible overhead when crawling through the entire directory.

File: src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts [hunk 2]
```diff
@@ -488,7 +488,10 @@ export class AnythingQuickAccessProvider extends PickerQuickAccessProvider<IAnyt
 
 	private createFileQueryCache(): FileQueryCacheState {
 		return new FileQueryCacheState(
-			cacheKey => this.fileQueryBuilder.file(this.contextService.getWorkspace().folders, this.getFileQueryOptions({ cacheKey })),
+			cacheKey => this.fileQueryBuilder.file(this.contextService.getWorkspace().folders, this.getFileQueryOptions({
+				cacheKey,
+				maxResults: AnythingQuickAccessProvider.MAX_RESULTS,
```
Comment (JacksonKearl) [inline src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts:493]: @roblourens I was seeing `maxResults` set to 0 which ended up overloading the quick picker.

File: src/vs/platform/files/browser/htmlFileSystemProvider.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/platform/files/browser/htmlFileSystemProvider.ts:368]: Not really having a better suggestion, but I would rather make `getHandle` public and then check on the returned type to be a directory. 

The entire way how to get handles in and out of the provider is a bit of a hack, as indicated by how the file dialog service uses it today:

https://github.com/microsoft/vscode/blob/17314f56bab57d945fb16cf30ca84b8539afadb7/src/vs/workbench/services/dialogs/browser/fileDialogService.ts#L24-L26

Comment (JacksonKearl) [reply src/vs/platform/files/browser/htmlFileSystemProvider.ts]: Sounds good, changed is so `getHandle` is now the public one. Saw that file service snippet, it was the basis for what I did here. 

It'd be cool to have the handles stored for reload, I wonder how the browsers would handle it. I was a bit surprised to see I could send the handle over to a webworker so easily.

Comment (bpasero) [reply src/vs/platform/files/browser/htmlFileSystemProvider.ts]: I think that is really the idea that you can persist and transfer these handles to use them in other contexts or later. However, you will have to call `queryPermission`: https://web.dev/file-system-access/#stored-file-or-directory-handles-and-permissions

File: src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts (hunk 2 above)
Comment (JacksonKearl) [reply src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts]: reverted this to go with a more local approach in the new file system provider

**Comment by joaomoreno** [comment]
Very cool!

**Pull request merged**